modules/nw-kuryr-cleanup.adoc

```diff
@@ -0,0 +1,208 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-kuryr-sdn.adoc
+
+:_content-type: PROCEDURE
+[id="nw-kuryr-cleanup_{context}"]
+= Cleaning up resources after migration
+
+After migration is done, there are necessary steps to clean up resources used by Kuryr. From high level perspective it will include:
+
+* Removing Kuryr finalizers from resources:
+* Removing Kuryr CRDs
+* Removing {rh-openstack-first} resources
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Install `jq` tool
+* Install Python interpreter
+* Install `openstacksdk` python package
+* Install `openstack` cli tool
+* Access to the underlying {rh-openstack}
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Find out cluster and Kuryr related identifiers, which will be used later:
++
+[source,terminal]
+----
+$ CLUSTERID=$(oc get infrastructure.config.openshift.io cluster -o=jsonpath='{.status.infrastructureName}')
+$ CLUSTERTAG="openshiftClusterID=${CLUSTERID}"
+$ ROUTERID=$(oc get kuryrnetwork -A --no-headers -o custom-columns=":status.routerId"|head -n 1)
+----
+
+. Create helper bash function, which will help out with removing finalizers
+from specified resources. It might look complicated, but it simply gets two
+parameters - first one is name of the resource, and the second is the
+finalizer to be removed. Than it will get all the desired resource from the
+cluster and replace it's definition with the same data excluding specified
+finalizer.
++
+[source,terminal]
+----
+$ function REMFIN {
```

Comment:
To reduce the chances of certain step being skipped or done wrongly, what do you think about having a list of resource in this function that we could iterate over and apply the necessary k8s actions and openstack actions that the specific resource requires?

Reply:
@gryf Are the steps of each resource removal supposed to happen in the order listed specified in the docs?
If yes, can we mention in the docs that the user must follow the order

Reply:
Ah, right. The order is important.

Reply:
Actually, the steps are presented as ordered list. Should we explicitly say, that order should be maintained? @maxwelldb WDY?

Reply:
Nah. Using ordered lists to indicate a particular order is understood in the docs. @gryf